packages/admin/cms-admin/src/common/routeMenu/useMenuForCurrentUser.ts

```diff
@@ -0,0 +1,30 @@
+import { MenuItemRouterLinkProps } from "@comet/admin";
+
+import { RouteMenu } from "./routeMenu.type";
+
+type Menu = {
+    menuItem: MenuItemRouterLinkProps;
+    hasSubMenu: boolean;
+    subMenu: {
+        menuItem: MenuItemRouterLinkProps;
+    }[];
+}[];
+
+export function useMenuForCurrentUser(items: RouteMenu, basePath: string): Menu {
```

Comment:
why did you decide for a hook instead of passing the user (while you pass the basePath)?

wouldn't a filter function (that can be used for routes + menu) (gets the user passed) simpler?
(the basePath could be added in the menu component)

Reply:
I created helper functions.